File: 2024/198. House Robber/Solution2.java
class Solution {
    public int rob(int[] nums) {
        if (nums.length < 2)
            return nums[0];
        int prev = nums[0], cur =  Math.max(nums[0], nums[1]);
        for (int i = 2; i < nums.length; i++) {
            int tmp = Math.max(cur, prev + nums[i]);
            prev = cur;
            cur = tmp;
        }
        return cur;
    }
}

File: 2025/53. Maximum Subarray/solution2.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        cur = max_sum = nums[0]
        for i in range(1, len(nums)):
            cur = max(cur + nums[i], nums[i])
            max_sum = max(max_sum, cur)
        return max_sum